File: mobile/android/SyncProxyAndroid/src/com/ford/syncV4/proxy/rpc/OnVehicleData.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;

import com.ford.syncV4.proxy.RPCNotification;
import com.ford.syncV4.proxy.constants.Names;
import com.ford.syncV4.proxy.rpc.enums.PRNDL;
import com.ford.syncV4.util.DebugTool;

public class OnVehicleData extends RPCNotification {

    public OnVehicleData() {
        super("OnVehicleData");
    }
    public OnVehicleData(Hashtable hash) {
        super(hash);
    }
    public void setGps(GPSData gps) {
    	if (gps != null) {
    		parameters.put(Names.gps, gps);
    	} else {
    		parameters.remove(Names.gps);
    	}
    }
    public GPSData getGps() {
    	Object obj = parameters.get(Names.gps);
        if (obj instanceof GPSData) {
            return (GPSData) obj;
        } else {
        	GPSData theCode = null;
            try {
                theCode = new GPSData((Hashtable) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.gps, e);
            }
            return theCode;
        }
    }
    public void setSpeed(Integer speed) {
    	if (speed != null) {
    		store.put(Names.speed, speed);
    	} else {
    		store.remove(Names.speed);
    	}
    }
    public Integer getSpeed() {
    	return (Integer) store.get(Names.speed);
    }
    public void setRpm(Integer rpm) {
    	if (rpm != null) {
    		store.put(Names.rpm, rpm);
    	} else {
    		store.remove(Names.rpm);
    	}
    }
    public Integer getRpm() {
    	return (Integer) store.get(Names.rpm);
    }
    public void setFuelLevel(Float fuelLevel) {
    	if (fuelLevel != null) {
    		store.put(Names.fuelLevel, fuelLevel);
    	} else {
    		store.remove(Names.fuelLevel);
    	}
    }
    public Float getFuelLevel() {
    	return (Float) store.get(Names.fuelLevel);
    }
    public void setAvgFuelEconomy(Float avgFuelEconomy) {
    	if (avgFuelEconomy != null) {
    		store.put(Names.avgFuelEconomy, avgFuelEconomy);
    	} else {
    		store.remove(Names.avgFuelEconomy);
    	}
    }
    public Float getAvgFuelEconomy() {
    	return (Float) store.get(Names.avgFuelEconomy);
    }
    public void setBatteryVoltage(Float batteryVoltage) {
    	if (batteryVoltage != null) {
    		store.put(Names.batteryVoltage, batteryVoltage);
    	} else {
    		store.remove(Names.batteryVoltage);
    	}
    }
    public Float getBatteryVoltage() {
    	return (Float) store.get(Names.batteryVoltage);
    }
    public void setExternalTemperature(Float externalTemperature) {
    	if (externalTemperature != null) {
    		store.put(Names.externalTemperature, externalTemperature);
    	} else {
    		store.remove(Names.externalTemperature);
    	}
    }
    public Float getExternalTemperature() {
    	return (Float) store.get(Names.externalTemperature);
    }
    public void setVin(String vin) {
    	if (vin != null) {
    		store.put(Names.vin, vin);
    	} else {
    		store.remove(Names.vin);
    	}
    }
    public String getVin() {
    	return (String) store.get(Names.vin);
    }
    public void setPrndl(PRNDL prndl) {
    	if (prndl != null) {
    		store.put(Names.prndl, prndl);
    	} else {
    		store.remove(Names.prndl);
    	}
    }
    public PRNDL getPrndl() {
        Object obj = store.get(Names.prndl);
        if (obj instanceof PRNDL) {
            return (PRNDL) obj;
        } else if (obj instanceof String) {
        	PRNDL theCode = null;
            try {
                theCode = PRNDL.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.prndl, e);
            }
            return theCode;
        }
        return null;
    }
    public void setTirePressure(TireStatus tirePressure) {
    	if (tirePressure != null) {
    		store.put(Names.tirePressure, tirePressure);
    	} else {
    		store.remove(Names.tirePressure);
    	}
    }
    public TireStatus getTirePressure() {
    	Object obj = store.get(Names.tirePressure);
        if (obj instanceof TireStatus) {
            return (TireStatus) obj;
        } else if (obj instanceof Hashtable) {
        	try {
        		return new TireStatus((Hashtable) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.tirePressure, e);
            }
        }
        return null;
    }
    public void setBatteryPackVoltage(Float batteryPackVoltage) {
    	if (batteryPackVoltage != null) {
    		store.put(Names.batteryPackVoltage, batteryPackVoltage);
    	} else {
    		store.remove(Names.batteryPackVoltage);
    	}
    }
    public Float getBatteryPackVoltage() {
    	return (Float) store.get(Names.batteryPackVoltage);
    }
    public void setBatteryPackCurrent(Float batteryPackCurrent) {
    	if (batteryPackCurrent != null) {
    		store.put(Names.batteryPackCurrent, batteryPackCurrent);
    	} else {
    		store.remove(Names.batteryPackCurrent);
    	}
    }
    public Float getBatteryPackCurrent() {
    	return (Float) store.get(Names.batteryPackCurrent);
    }
    public void setBatteryPackTemperature(Integer batteryPackTemperature) {
    	if (batteryPackTemperature != null) {
    		store.put(Names.batteryPackTemperature, batteryPackTemperature);
    	} else {
    		store.remove(Names.batteryPackTemperature);
    	}
    }
    public Integer getBatteryPackTemperature() {
    	return (Integer) store.get(Names.batteryPackTemperature);
    }

    public void setHeadLampStatus(HeadLampStatus headLampStatus) {
        if (headLampStatus != null) {
            parameters.put(Names.headLampStatus, headLampStatus);
        } else {
            parameters.remove(Names.headLampStatus);
        }
    }

    public HeadLampStatus getHeadLampStatus() {
        Object obj = parameters.get(Names.headLampStatus);
        if (obj instanceof HeadLampStatus) {
            return (HeadLampStatus) obj;
        } else if (obj instanceof Hashtable) {
            try {
                return new HeadLampStatus((Hashtable) obj);
            } catch (Exception e) {
                DebugTool.logError(
                        "Failed to parse " + getClass().getSimpleName() + "." +
                                Names.headLampStatus, e);
            }
        }
        return null;
    }

    public void setEngineTorque(Integer engineTorque) {
    	if (engineTorque != null) {
    		store.put(Names.engineTorque, engineTorque);
    	} else {
    		store.remove(Names.engineTorque);
    	}
    }
    public Integer getEngineTorque() {
    	return (Integer) store.get(Names.engineTorque);
    }
    public void setOdometer(Integer odometer) {
    	if (odometer != null) {
    		store.put(Names.odometer, odometer);
    	} else {
    		store.remove(Names.odometer);
    	}
    }
    public Integer getOdometer() {
    	return (Integer) store.get(Names.odometer);
    }
    public void setTripOdometer(Integer tripOdometer) {
    	if (tripOdometer != null) {
    		store.put(Names.tripOdometer, tripOdometer);
    	} else {
    		store.remove(Names.tripOdometer);
    	}
    }
    public Integer getTripOdometer() {
    	return (Integer) store.get(Names.tripOdometer);
    }
    public void setSatRadioESN(String satRadioESN) {
    	if (satRadioESN != null) {
    		store.put(Names.satRadioESN, satRadioESN);
    	} else {
    		store.remove(Names.satRadioESN);
    	}
    }
    public String getSatRadioESN() {
    	return (String) store.get(Names.satRadioESN);
    }

    public void setSteeringWheelAngle(Float steeringWheelAngle) {
        if (steeringWheelAngle != null) {
            parameters.put(Names.steeringWheelAngle, steeringWheelAngle);
        } else {
            parameters.remove(Names.steeringWheelAngle);
        }
    }

    public Float getSteeringWheelAngle() {
        final Object o = parameters.get(Names.steeringWheelAngle);
        if (o instanceof Float) {
            return (Float) o;
        }
        if (o instanceof Double) {
            return ((Double) o).floatValue();
        }
        return null;
    }

    public void setECallInfo(ECallInfo eCallInfo) {
        if (eCallInfo != null) {
            parameters.put(Names.eCallInfo, eCallInfo);
        } else {
            parameters.remove(Names.eCallInfo);
        }
    }

    public ECallInfo getECallInfo() {
        Object obj = parameters.get(Names.eCallInfo);
        if (obj instanceof ECallInfo) {
            return (ECallInfo) obj;
        } else if (obj instanceof Hashtable) {
            try {
                return new ECallInfo((Hashtable) obj);
            } catch (Exception e) {
                DebugTool.logError(
                        "Failed to parse " + getClass().getSimpleName() + "." +
                                Names.eCallInfo, e);
            }
        }
        return null;
    }

    public void setAirbagStatus(AirbagStatus airbagStatus) {
        if (airbagStatus != null) {
            parameters.put(Names.airbagStatus, airbagStatus);
        } else {
            parameters.remove(Names.airbagStatus);
        }
    }

    public AirbagStatus getAirbagStatus() {
        Object obj = parameters.get(Names.airbagStatus);
        if (obj instanceof AirbagStatus) {
            return (AirbagStatus) obj;
        } else if (obj instanceof Hashtable) {
            try {
                return new AirbagStatus((Hashtable) obj);
            } catch (Exception e) {
                DebugTool.logError(
                        "Failed to parse " + getClass().getSimpleName() + "." +
                                Names.airbagStatus, e);
            }
        }
        return null;
    }

    public void setEmergencyEvent(EmergencyEvent emergencyEvent) {
        if (emergencyEvent != null) {
            parameters.put(Names.emergencyEvent, emergencyEvent);
        } else {
            parameters.remove(Names.emergencyEvent);
        }
    }

    public EmergencyEvent getEmergencyEvent() {
        Object obj = parameters.get(Names.emergencyEvent);
        if (obj instanceof EmergencyEvent) {
            return (EmergencyEvent) obj;
        } else if (obj instanceof Hashtable) {
            try {
                return new EmergencyEvent((Hashtable) obj);
            } catch (Exception e) {
                DebugTool.logError(
                        "Failed to parse " + getClass().getSimpleName() + "." +
                                Names.emergencyEvent, e);
            }
        }
        return null;
    }

    public void setClusterModeStatus(ClusterModeStatus clusterModeStatus) {
        if (clusterModeStatus != null) {
            parameters.put(Names.clusterModeStatus, clusterModeStatus);
        } else {
            parameters.remove(Names.clusterModeStatus);
        }
    }

    public ClusterModeStatus getClusterModeStatus() {
        Object obj = parameters.get(Names.clusterModeStatus);
        if (obj instanceof ClusterModeStatus) {
            return (ClusterModeStatus) obj;
        } else if (obj instanceof Hashtable) {
            try {
                return new ClusterModeStatus((Hashtable) obj);
            } catch (Exception e) {
                DebugTool.logError(
                        "Failed to parse " + getClass().getSimpleName() + "." +
                                Names.clusterModeStatus, e);
            }
        }
        return null;
    }

    public void setMyKey(MyKey myKey) {
        if (myKey != null) {
            parameters.put(Names.myKey, myKey);
        } else {
            parameters.remove(Names.myKey);
        }
    }

    public MyKey getMyKey() {
        Object obj = parameters.get(Names.myKey);
        if (obj instanceof MyKey) {
            return (MyKey) obj;
        } else if (obj instanceof Hashtable) {
            try {
                return new MyKey((Hashtable) obj);
            } catch (Exception e) {
                DebugTool.logError(
                        "Failed to parse " + getClass().getSimpleName() + "." +
                                Names.myKey, e);
            }
        }
        return null;
    }
}

File: src/components/JSONHandler/src/ALRPCObjectsImpl/V2/OnVehicleData.cpp
#include "../include/JSONHandler/ALRPCObjects/V2/OnVehicleData.h"
#include "OnVehicleDataMarshaller.h"
#include "../include/JSONHandler/ALRPCObjects/V2/Marshaller.h"
#include "GPSDataMarshaller.h"
#include "PRNDLMarshaller.h"
#include "TireStatusMarshaller.h"

#define PROTOCOL_VERSION	2


/*
  interface	Ford Sync RAPI
  version	2.0O
  date		2012-11-02
  generated at	Thu Dec 13 14:18:29 2012
  source stamp	Thu Dec 13 14:18:27 2012
  author	robok0der
*/

using namespace NsAppLinkRPCV2;
OnVehicleData& OnVehicleData::operator =(const OnVehicleData& c)
{
  gps= c.gps ? new GPSData(c.gps[0]) : 0;
  speed= c.speed ? new float(c.speed[0]) : 0;
  rpm= c.rpm ? new unsigned int(c.rpm[0]) : 0;
  fuelLevel= c.fuelLevel ? new float(c.fuelLevel[0]) : 0;
  avgFuelEconomy= c.avgFuelEconomy ? new float(c.avgFuelEconomy[0]) : 0;
  batteryVoltage= c.batteryVoltage ? new float(c.batteryVoltage[0]) : 0;
  externalTemperature= c.externalTemperature ? new float(c.externalTemperature[0]) : 0;
  vin= c.vin ? new std::string(c.vin[0]) : 0;
  prndl= c.prndl ? new PRNDL(c.prndl[0]) : 0;
  tirePressure= c.tirePressure ? new TireStatus(c.tirePressure[0]) : 0;
  batteryPackVoltage= c.batteryPackVoltage ? new float(c.batteryPackVoltage[0]) : 0;
  batteryPackCurrent= c.batteryPackCurrent ? new float(c.batteryPackCurrent[0]) : 0;
  batteryPackTemperature= c.batteryPackTemperature ? new int(c.batteryPackTemperature[0]) : 0;
  engineTorque= c.engineTorque ? new int(c.engineTorque[0]) : 0;
  odometer= c.odometer ? new unsigned int(c.odometer[0]) : 0;
  tripOdometer= c.tripOdometer ? new unsigned int(c.tripOdometer[0]) : 0;
  genericbinary= c.genericbinary ? new std::string(c.genericbinary[0]) : 0;

  return *this;
}


OnVehicleData::~OnVehicleData(void)
{
  if(gps)
    delete gps;
  if(speed)
    delete speed;
  if(rpm)
    delete rpm;
  if(fuelLevel)
    delete fuelLevel;
  if(avgFuelEconomy)
    delete avgFuelEconomy;
  if(batteryVoltage)
    delete batteryVoltage;
  if(externalTemperature)
    delete externalTemperature;
  if(vin)
    delete vin;
  if(prndl)
    delete prndl;
  if(tirePressure)
    delete tirePressure;
  if(batteryPackVoltage)
    delete batteryPackVoltage;
  if(batteryPackCurrent)
    delete batteryPackCurrent;
  if(batteryPackTemperature)
    delete batteryPackTemperature;
  if(engineTorque)
    delete engineTorque;
  if(odometer)
    delete odometer;
  if(tripOdometer)
    delete tripOdometer;
  if(genericbinary)
    delete genericbinary;
}


OnVehicleData::OnVehicleData(const OnVehicleData& c) : NsAppLinkRPC::ALRPCMessage(c)
{
  *this=c;
}


bool OnVehicleData::checkIntegrity(void)
{
  return OnVehicleDataMarshaller::checkIntegrity(*this);
}


OnVehicleData::OnVehicleData(void) : NsAppLinkRPC::ALRPCMessage(PROTOCOL_VERSION),
      gps(0),
    speed(0),
    rpm(0),
    fuelLevel(0),
    avgFuelEconomy(0),
    batteryVoltage(0),
    externalTemperature(0),
    vin(0),
    prndl(0),
    tirePressure(0),
    batteryPackVoltage(0),
    batteryPackCurrent(0),
    batteryPackTemperature(0),
    engineTorque(0),
    odometer(0),
    tripOdometer(0),
    genericbinary(0)
{
}



bool OnVehicleData::set_gps(const GPSData& gps_)
{
  if(!GPSDataMarshaller::checkIntegrityConst(gps_))   return false;
  delete gps;
  gps=0;

  gps=new GPSData(gps_);
  return true;
}

void OnVehicleData::reset_gps(void)
{
  if(gps)
    delete gps;
  gps=0;
}

bool OnVehicleData::set_speed(float speed_)
{
  if(speed_>500)  return false;
  if(speed_<-100)  return false;
  delete speed;
  speed=0;

  speed=new float(speed_);
  return true;
}

void OnVehicleData::reset_speed(void)
{
  if(speed)
    delete speed;
  speed=0;
}

bool OnVehicleData::set_rpm(unsigned int rpm_)
{
  if(rpm_>20000)  return false;
  delete rpm;
  rpm=0;

  rpm=new unsigned int(rpm_);
  return true;
}

void OnVehicleData::reset_rpm(void)
{
  if(rpm)
    delete rpm;
  rpm=0;
}

bool OnVehicleData::set_fuelLevel(float fuelLevel_)
{
  if(fuelLevel_>100)  return false;
  if(fuelLevel_<0)  return false;
  delete fuelLevel;
  fuelLevel=0;

  fuelLevel=new float(fuelLevel_);
  return true;
}

void OnVehicleData::reset_fuelLevel(void)
{
  if(fuelLevel)
    delete fuelLevel;
  fuelLevel=0;
}

bool OnVehicleData::set_avgFuelEconomy(float avgFuelEconomy_)
{
  if(avgFuelEconomy_>7000)  return false;
  if(avgFuelEconomy_<0)  return false;
  delete avgFuelEconomy;
  avgFuelEconomy=0;

  avgFuelEconomy=new float(avgFuelEconomy_);
  return true;
}

void OnVehicleData::reset_avgFuelEconomy(void)
{
  if(avgFuelEconomy)
    delete avgFuelEconomy;
  avgFuelEconomy=0;
}

bool OnVehicleData::set_batteryVoltage(float batteryVoltage_)
{
  if(batteryVoltage_>26)  return false;
  if(batteryVoltage_<0)  return false;
  delete batteryVoltage;
  batteryVoltage=0;

  batteryVoltage=new float(batteryVoltage_);
  return true;
}

void OnVehicleData::reset_batteryVoltage(void)
{
  if(batteryVoltage)
    delete batteryVoltage;
  batteryVoltage=0;
}

bool OnVehicleData::set_externalTemperature(float externalTemperature_)
{
  if(externalTemperature_>100)  return false;
  if(externalTemperature_<-40)  return false;
  delete externalTemperature;
  externalTemperature=0;

  externalTemperature=new float(externalTemperature_);
  return true;
}

void OnVehicleData::reset_externalTemperature(void)
{
  if(externalTemperature)
    delete externalTemperature;
  externalTemperature=0;
}

bool OnVehicleData::set_vin(const std::string& vin_)
{
  if(vin_.length()>17)  return false;
  delete vin;
  vin=0;

  vin=new std::string(vin_);
  return true;
}

void OnVehicleData::reset_vin(void)
{
  if(vin)
    delete vin;
  vin=0;
}

bool OnVehicleData::set_prndl(const PRNDL& prndl_)
{
  if(!PRNDLMarshaller::checkIntegrityConst(prndl_))   return false;
  delete prndl;
  prndl=0;

  prndl=new PRNDL(prndl_);
  return true;
}

void OnVehicleData::reset_prndl(void)
{
  if(prndl)
    delete prndl;
  prndl=0;
}

bool OnVehicleData::set_tirePressure(const TireStatus& tirePressure_)
{
  if(!TireStatusMarshaller::checkIntegrityConst(tirePressure_))   return false;
  delete tirePressure;
  tirePressure=0;

  tirePressure=new TireStatus(tirePressure_);
  return true;
}

void OnVehicleData::reset_tirePressure(void)
{
  if(tirePressure)
    delete tirePressure;
  tirePressure=0;
}

bool OnVehicleData::set_batteryPackVoltage(float batteryPackVoltage_)
{
  if(batteryPackVoltage_>1000)  return false;
  if(batteryPackVoltage_<-1000)  return false;
  delete batteryPackVoltage;
  batteryPackVoltage=0;

  batteryPackVoltage=new float(batteryPackVoltage_);
  return true;
}

void OnVehicleData::reset_batteryPackVoltage(void)
{
  if(batteryPackVoltage)
    delete batteryPackVoltage;
  batteryPackVoltage=0;
}

bool OnVehicleData::set_batteryPackCurrent(float batteryPackCurrent_)
{
  if(batteryPackCurrent_>1000)  return false;
  if(batteryPackCurrent_<-1000)  return false;
  delete batteryPackCurrent;
  batteryPackCurrent=0;

  batteryPackCurrent=new float(batteryPackCurrent_);
  return true;
}

void OnVehicleData::reset_batteryPackCurrent(void)
{
  if(batteryPackCurrent)
    delete batteryPackCurrent;
  batteryPackCurrent=0;
}

bool OnVehicleData::set_batteryPackTemperature(int batteryPackTemperature_)
{
  if(batteryPackTemperature_>250)  return false;
  delete batteryPackTemperature;
  batteryPackTemperature=0;

  batteryPackTemperature=new int(batteryPackTemperature_);
  return true;
}

void OnVehicleData::reset_batteryPackTemperature(void)
{
  if(batteryPackTemperature)
    delete batteryPackTemperature;
  batteryPackTemperature=0;
}

bool OnVehicleData::set_engineTorque(int engineTorque_)
{
  if(engineTorque_>55000)  return false;
  delete engineTorque;
  engineTorque=0;

  engineTorque=new int(engineTorque_);
  return true;
}

void OnVehicleData::reset_engineTorque(void)
{
  if(engineTorque)
    delete engineTorque;
  engineTorque=0;
}

bool OnVehicleData::set_odometer(unsigned int odometer_)
{
  if(odometer_>2000000)  return false;
  delete odometer;
  odometer=0;

  odometer=new unsigned int(odometer_);
  return true;
}

void OnVehicleData::reset_odometer(void)
{
  if(odometer)
    delete odometer;
  odometer=0;
}

bool OnVehicleData::set_tripOdometer(unsigned int tripOdometer_)
{
  if(tripOdometer_>2000000)  return false;
  delete tripOdometer;
  tripOdometer=0;

  tripOdometer=new unsigned int(tripOdometer_);
  return true;
}

void OnVehicleData::reset_tripOdometer(void)
{
  if(tripOdometer)
    delete tripOdometer;
  tripOdometer=0;
}

bool OnVehicleData::set_genericbinary(const std::string& genericbinary_)
{
  if(genericbinary_.length()>500)  return false;
  delete genericbinary;
  genericbinary=0;

  genericbinary=new std::string(genericbinary_);
  return true;
}

void OnVehicleData::reset_genericbinary(void)
{
  if(genericbinary)
    delete genericbinary;
  genericbinary=0;
}




const GPSData* OnVehicleData::get_gps(void) const 
{
  return gps;
}

const float* OnVehicleData::get_speed(void) const 
{
  return speed;
}

const unsigned int* OnVehicleData::get_rpm(void) const 
{
  return rpm;
}

const float* OnVehicleData::get_fuelLevel(void) const 
{
  return fuelLevel;
}

const float* OnVehicleData::get_avgFuelEconomy(void) const 
{
  return avgFuelEconomy;
}

const float* OnVehicleData::get_batteryVoltage(void) const 
{
  return batteryVoltage;
}

const float* OnVehicleData::get_externalTemperature(void) const 
{
  return externalTemperature;
}

const std::string* OnVehicleData::get_vin(void) const 
{
  return vin;
}

const PRNDL* OnVehicleData::get_prndl(void) const 
{
  return prndl;
}

const TireStatus* OnVehicleData::get_tirePressure(void) const 
{
  return tirePressure;
}

const float* OnVehicleData::get_batteryPackVoltage(void) const 
{
  return batteryPackVoltage;
}

const float* OnVehicleData::get_batteryPackCurrent(void) const 
{
  return batteryPackCurrent;
}

const int* OnVehicleData::get_batteryPackTemperature(void) const 
{
  return batteryPackTemperature;
}

const int* OnVehicleData::get_engineTorque(void) const 
{
  return engineTorque;
}

const unsigned int* OnVehicleData::get_odometer(void) const 
{
  return odometer;
}

const unsigned int* OnVehicleData::get_tripOdometer(void) const 
{
  return tripOdometer;
}

const std::string* OnVehicleData::get_genericbinary(void) const 
{
  return genericbinary;
}

